doc/aaa/SONiC RBAC.md

```diff
@@ -0,0 +1,486 @@
+    * *resource* : The xpath associated with the resource being accessed. This is a string.
+    * *tenant* : The tenant for which the resource partitioning is being done. This is a string.
+    * *instances* : The instances of the *resource* allocated to this *tenant*. This is a list of instances.
+  The TenantTable is keyed on <***resource, tenant***>
+
+### 3.2.2 APP DB
+N/A
+
+### 3.2.3 STATE DB
+N/A
+
+### 3.2.4 ASIC DB
+N/A
+
+### 3.2.5 COUNTER DB
+N/A
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+N/A
+
+### 3.3.2 Other Process
+N/A
+
+## 3.4 SyncD
+
+N/A. HAM is going to be the single source of truth, and it will take care of
+synchronizing the User Table and /etc/passwd
+
+## 3.5 SAI
+N/A
+
+## 3.6 User Interface
+### 3.6.1 Data Models
+
+TBD from developer
+(TODO/DELL)
+
+### 3.6.2 CLI
+#### 3.6.2.1 Configuration Commands for User Management
+Users may be managed via Linux tools like `sonic-useradd`, `sonic-usermod`, `passwd`, etc. They may also be managed via configuration.
+
+##### username
+`username <name> password <password-string> role <role-string>` -- Configures a user on the system with a given name, password, and role.
+* **name** is a text string of 1-32 alphanumeric characters
+* **password-string** is a text string of 1-32 alphanumeric characters
+* **role-string** is a text string consisting of a role name. In the initial release, the user is recommended to use "admin" and "operator" roles, as other roles will not be supported. A text string is desired instead of keywords so that in the future, more roles may be implemented and expanded.
+* Configuring another a user with the same **name** should result in modification of the existing user.
+
+`no username <name>` -- Deletes a user from the system.
+* **name** is a text string of 1-32 alphanumeric characters
+
+##### aaa authentication
+```
+aaa authentication {rest-server | gnmi-server} {[password] [token] [certificate]}
+aaa authentication {rest-server | gnmi-server} certificate no-authorization
+no aaa authentication {rest-server | gnmi-server}
+```
+
+The `certificate no-authorization` mode will force users to authenticate via
+certificate, however, there will be no role lookup performed. All authenticated
+users will be treated as having full administrator access.
+
+The `no aaa ...` command will disable authentication altogether, and will allow
+all users to execute admin level commands via the REST/gNMI interfaces. The CLI
+however, will still require a valid password based login, due to it being
+accessed via SSH.
+
+##### userrole
+`userrole <name>` - Creates a new custom role and enters userrole config to
+enable/disable individual features.
+`(config-userrole)# feature <feat-name> {read-only | read-write}` - Enable
```

Comment:
How does user know about the correct feature-name? Is there a command to list out the feature-name which can be configured for read-only or read-write